compiler/src/dotty/tools/repl/ReplCompiler.scala

```diff
@@ -26,39 +27,43 @@ import scala.collection.mutable
 /** This subclass of `Compiler` replaces the appropriate phases in order to
  *  facilitate the REPL
  *
- *  Specifically it replaces the front end with `REPLFrontEnd`, and adds a
- *  custom subclass of `GenBCode`. The custom `GenBCode`, `REPLGenBCode`, works
- *  in conjunction with a specialized class loader in order to load virtual
- *  classfiles.
+ *  Specifically it replaces the front end with `REPLFrontEnd`.
  */
 class ReplCompiler extends Compiler {
-  override protected def frontendPhases: List[List[Phase]] =
-    Phases.replace(classOf[FrontEnd], _ => new REPLFrontEnd :: Nil, super.frontendPhases)
```

Comment:
I think keeping Phases.replace would be cleaner, it just needs to be passed a list of list of phases as a replacement.

Reply:
That would be trickier. I need to replace `FrontEnd` by `ReplFrontend`, remove `sbt.ExtractDependencies` and `sbt.ExtractAPI` and finally add `CollectTopLevelImports` which needs to run after `ReplFrontEnd` but in its own group

Reply:
Ah indeed, not worth it then